drawBot/context/svgContext.py

```diff
@@ -462,17 +465,20 @@ def _image(self, path, xy, alpha, pageNumber):
             url = AppKit.NSURL.URLWithString_(path)
         else:
             url = AppKit.NSURL.fileURLWithPath_(path)
-        im = AppKit.NSImage.alloc().initByReferencingURL_(url)
-        w, h = im.size()
+        image = AppKit.NSImage.alloc().initByReferencingURL_(url)
+        w, h = image.size()
+        imageRep = _makeBitmapImageRep(image)
+        imageData = imageRep.representationUsingType_properties_(AppKit.NSPNGFileType, None)
```

Comment:
I think if the source image is PNG or JPG (or GIF?) the original data should be embedded. Other formats could be converted to PNG like you do here.